docs/DeveloperGuide.adoc

```diff
@@ -122,125 +119,647 @@ image::ModelClassDiagram.png[]
+.Activity Diagram for `mod list` Command
+image::ModuleListActivityDiagram.png[]
+
+==== Design Considerations
+
+===== Aspect: Support for editing module code
+* **Alternative 1 (current choice):** Allow users to edit the module code of a module.
+** Pros: More flexibility for users.
+** Cons: More complex implementation as the classes, tasks and facilitators all store module codes and have to be edited too.
+* **Alternative 2:** Allow users to only edit the description of a module.
+** Pros: Easier to implement.
+** Cons: More rigid for users.
+
+=== Class Management Feature
+The class feature manages the classes in Mod Manager and is represented by the `Lesson` class.
+A class has a `ModuleCode`, `LessonType`, `day` which is a `DayOfWeek` object, `startTime`, `endTime` which are `LocalTime` objects and `venue` which is a `String`.
+
+It supports the following operations:
+
+* add - Adds a class to Mod Manager.
+* list - Lists all classes in Mod Manager.
+* edit - Edits a class in Mod Manager.
+* delete - Deletes a class in Mod Manager.
+
+==== Implementation Details
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+===== Adding a class
+The add class command allows user to add a class to ModManager. This feature is facilitated by `LessonCommandParser`, `LessonAddCommandParser` and `LessonAddCommand`. The operation is exposed in the `Model` interface as `Model#addLesson()`.
 
-image::UndoRedoState4.png[]
+Given below is an example usage scenario and how the lesson add mechanism behaves at each step.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+1. The user executes the lesson add command and provides the module code, lesson type, day, start time, end time and venue of the lesson to be added.
+2. `LessonAddCommandParser` creates a new `Lesson`, then a new `LessonAddCommand`.
+3. `LogicManager` executes the `LessonAddCommand`.
+4. `ModManager` adds the `Lesson` to `LessonList`.
 
-image::UndoRedoState5.png[]
+The following sequence diagram shows how the lesson add command works:
 
-The following activity diagram summarizes what happens when a user executes a new command:
+.Sequence Diagram for `class add` Command
+image::LessonAddSequenceDiagram.png[]
```

Comment:
You can use a generic command execute(...) instead of a specific example. This will help save some space in your diagram, which you can make bigger for better readability.